src/Database/Redis/Cluster.hs

```diff
@@ -121,48 +122,56 @@ instance Exception CrossSlotException
 instance Exception NoNodeException
 
-connect :: (Host -> CC.PortID -> Maybe Int -> IO CC.ConnectionContext) -> [CMD.CommandInfo] -> MVar ShardMap -> Maybe Int -> Bool -> (NodeConnection -> IO ShardMap) -> IO Connection
-connect withAuth commandInfos shardMapVar timeoutOpt isReadOnly refreshShardMap = do
+connect :: (Host -> CC.PortID -> Maybe Int -> IO CC.ConnectionContext) -> [CMD.CommandInfo] -> MVar ShardMap -> Maybe Int -> Bool -> (NodeConnection -> IO ShardMap) -> Time.NominalDiffTime -> Int -> IO Connection
+connect withAuth commandInfos shardMapVar timeoutOpt isReadOnly _refreshShardMap idleTime maxResources = do
         shardMap <- readMVar shardMapVar
+        -- shardMap <- modifyMVar shardMapVar (\(ShardMap intMap) -> case (IntMap.toList intMap) of
+        --     (first:rest) -> return (ShardMap (IntMap.fromList rest), ShardMap (IntMap.fromList [first]))
+        --     _ -> error "Invalid ShardMap")
         stateVar <- newMVar $ Pending []
         pipelineVar <- newMVar $ Pipeline stateVar
-        (eNodeConns, shouldRetry) <- nodeConnections shardMap
+        -- (eNodeConns, shouldRetry) <- nodeConnections shardMap
+        nodeConns <- nodeConnections shardMap
+        -- print $ map (\(k, _) -> k) $ HM.toList eNodeConns
+        -- return eNodeConns
         -- whenever one of the node connection is not established,
         -- will refresh the slots and retry node connections.
         -- This would handle fail over, IP change use cases.
-        nodeConns <-
-          if shouldRetry
-            then if not (HM.null eNodeConns)
-                    then do
-                      newShardMap <- refreshShardMap (head $ HM.elems eNodeConns)
-                      refreshShardMapVar newShardMap
-                      simpleNodeConnections newShardMap
-                    else
-                      throwIO NoNodeException
-            else
-              return eNodeConns
-        return $ Connection nodeConns pipelineVar shardMapVar (CMD.newInfoMap commandInfos) isReadOnly where
-    simpleNodeConnections :: ShardMap -> IO (HM.HashMap NodeID NodeConnection)
-    simpleNodeConnections shardMap = HM.fromList <$> mapM connectNode (nub $ nodes shardMap)
-    nodeConnections :: ShardMap -> IO (HM.HashMap NodeID NodeConnection, Bool)
+        -- nodeConns <-
+        --   if shouldRetry
+        --     then if not (HM.null eNodeConns)
+        --             then do
+        --               newShardMap <- refreshShardMap (head $ HM.elems eNodeConns)
+        --               refreshShardMapVar newShardMap
+        --               simpleNodeConnections newShardMap
+        --             else
+        --               throwIO NoNodeException
+        --     else
+        --       return eNodeConns
+        let connection = Connection nodeConns pipelineVar shardMapVar (CMD.newInfoMap commandInfos) isReadOnly
+        return connection
+    where
+    -- simpleNodeConnections :: ShardMap -> IO (HM.HashMap NodeID NodeConnection)
```

Comment:
@shashitnak , can we remove these old commented code ? Doesn't look clean.

Reply:
@ishan-juspay removed them while rebasing to the new master